src/js/activities/create.js

```diff
@@ -6,25 +6,123 @@ var app = angular.module("confluente");
+    };
+
     // function called when new activity is submitted
     $scope.submit = function () {
         $scope.loading = true;
-
-        // create new activity from variables as put on the $scope by the form
-        activities.create({
+        // constructing standard activity object
+        var act = {
             name: $scope.name,
             description: $scope.description,
             organizer: $scope.organizer,
             date: $scope.date,
             startTime: $scope.startTime,
             endTime: $scope.endTime,
             location: $scope.location,
-            approved: true
-        }).then(function (result) {
+            participationFee: $scope.participationFee,
+            approved: true,
+            canSubscribe: $scope.canSubscribe,
+        };
+
+        // Checking required fields
+        $scope.empty = !$scope.name || !$scope.description || !$scope.organizer;
+
+        // Adding form to activity object if members can subscribe
+        if ($scope.canSubscribe) {
+            // Format output correctly
+            var allDescriptions = [];
+            var allTypes = [];
+            var allOptions = [];
+            var allRequired = [];
+
+            $scope.inputs.forEach(function (dataObj) {
+                allDescriptions.push(dataObj.fullQuestion);
+                allTypes.push(dataObj.type);
+
+                // SQlite database can't handle strings therefore lists are stored as , seperated lists and ; seperated lists
+                var optionString = dataObj.options[0];
+                for (var i = 1; i < dataObj.options.length; i++) {
+                    optionString += ";" + dataObj.options[i];
+                }
+                allOptions.push(optionString);
+
+                // check whether the question actually has a question
+                allRequired.push(dataObj.required);
+                if (!dataObj.fullQuestion || dataObj.fullQuestion === "") {
+                    $scope.empty = true;
+                }
+
+                // Check whether choices of multiple choice questions are empty
+                if (dataObj.type !== "☰ text" && dataObj.type !== "name" && dataObj.type !== "TU/e email") {
+                    for (var i = 0; i < dataObj.options.length; i++) {
+                        if (dataObj.options[i] === "" || !dataObj.options) $scope.empty = true;
+                    }
+                }
+            });
+
+            act.numberOfQuestions = allDescriptions.length;
+            act.typeOfQuestion = allTypes;
+            act.questionDescriptions = allDescriptions;
+            act.options = allOptions;
+            act.required = allRequired;
+            act.subscriptionDeadline = $scope.deadline.subscriptionDeadline;
+        }
+
+        // If any required field is empty than do not accept the activity
+        if ($scope.empty) {
+            $scope.loading = false;
+            return alert("One of your field is empty!");
```

Comment:
field => fields, maybe slightly more informative: "Not all required fields have been filled in"